Libraries/src/Amazon.Lambda.AspNetCoreServer/APIGatewayProxyFunction.cs

```diff
@@ -135,6 +102,40 @@ private protected override void InternalCustomResponseExceptionHandling(HostingA
         /// <param name="lambdaContext"></param>
         protected override void MarshallRequest(InvokeFeatures features, APIGatewayProxyRequest apiGatewayRequest, ILambdaContext lambdaContext)
         {
+            {
+                var authFeatures = (IHttpAuthenticationFeature) features;
```

Comment:
I agree it is weird but at this point I'm not ready to change the design. I actually think I got this from the ASP.NET Core repo back in the 1.0 days because I wouldn't have thought to do this way.